pkg/discovery/dtofactory/node_group_entity_dto_builder.go

```diff
@@ -55,6 +55,16 @@ func (builder *nodeGroupEntityDTOBuilder) BuildEntityDTOs() ([]*proto.EntityDTO,
 			continue
 		}
 
+		// Filling in ConnectedEntity field to reflect the mapping between the nodegroup and node
+		for nodeUID := range nodeLst {
+			connectedEntityID := nodeUID
```

Comment:
@irfanurrehman  the `nodeLst` is actually using the `node.UID`,  I shouldn't have used `nodeName` as the varaible name here.